edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/MetricsFileStorage.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class MetricsFileStorage : IMetricsStorage
+    {
+        readonly ISystemTime systemTime;
+
+        string directory;
+
+        public MetricsFileStorage(string directory, ISystemTime systemTime = null)
+        {
+            this.directory = Preconditions.CheckNotNull(directory, nameof(directory));
+            this.systemTime = systemTime ?? SystemTime.Instance;
+        }
+
+        public void WriteData(string data)
+        {
+            Directory.CreateDirectory(this.directory);
```

Comment:
this may throw exception; please handle it.

Reply:
There is no way to gracefully handle the write failing. It has to throw an exception to notify the caller it failed. In this case, the exception is logged by the periodic task.